File: fromJAVAtoPYTHON/04_OOP/07.Adding_nstance_methods/Book.py
class Book:
    # constructor
    def __init__(self, name, copies) -> None:
        self.name2 = name
        self.copies2 = copies

    #toString
    def __repr__(self) -> str:
        return repr((self.name2, self.copies2))

    def increase_copies(self, how_much):
        self.copies2 += how_much

    def decrease_copies(self, how_much):
        self.copies2 -= how_much

#instances
book2 = Book('John', 200)
book2.increase_copies(50
)
book3 = Book('Mike', 15)
book3.decrease_copies(10)

print(book2)
print(book3)

File: fromJAVAtoPYTHON/04_OOP/07.Adding_nstance_methods/Book.java
public class Book {
    String name;
    int copies;

    Book(String name, int copies) {
        this.name = name;
        this.copies = copies;
    }

    void increaseCopies(int howMuch) {
        this.copies += howMuch;
    }

    void decreaseCopies(int howMuch) {
        this.copies -= howMuch;
    }

    public String toString() {
        return "(" + name + ", " + copies + ")";
    }

    public static void main(String[] args) {
        Book book1 = new Book("John", 200);
        Book book2 = new Book("Mike", 15);

        System.out.println(book1);
        System.out.println(book2);
    }
}